presto-product-tests/src/main/java/com/facebook/presto/tests/SqlCancelTests.java

```diff
@@ -0,0 +1,194 @@
+import com.teradata.tempto.query.QueryResult;
+import io.airlift.http.client.HttpStatus;
+import io.airlift.http.client.Response;
+import io.airlift.log.Logger;
+import org.testng.annotations.Test;
+
+import java.sql.SQLException;
+import java.util.concurrent.Callable;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.Future;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+
+import static com.facebook.presto.tests.TestGroups.CANCEL_QUERY;
+import static com.teradata.tempto.assertions.QueryAssert.Row.row;
+import static com.teradata.tempto.assertions.QueryAssert.assertThat;
+import static com.teradata.tempto.query.QueryExecutor.query;
+import static java.lang.String.format;
+
+public class SqlCancelTests
+        extends ProductTest
+{
+    private static String tableName = "table_name";
+
+    @BeforeTestWithContext
+    @AfterTestWithContext
+    public void dropTestTables()
+    {
+        try {
+            query(format("DROP TABLE IF EXISTS %s", tableName));
+        }
+        catch (Exception e) {
+            Logger.get(getClass()).warn(e, "failed to drop table");
+        }
+    }
+
+    private class executeQuery
+            implements Callable<Boolean>
+    {
+        private String sql;
+
+        public executeQuery(String sql)
+        {
+            this.sql = sql;
+        }
+
+        public Boolean call()
+        {
+            try {
+                query(sql);
+                return false;
+            }
+            catch (QueryExecutionException e) {
+                return true;
+            }
+        }
+    }
+
+    private class cancelQuery
+            implements Callable<Response>
+    {
+        private String sqlStatement;
+
+        public cancelQuery(String sqlStatement)
+        {
+            this.sqlStatement = sqlStatement;
+        }
+
+        public Response call()
+        {
+            try {
+                Thread.sleep(10000);
+            }
+            catch (InterruptedException e) {
+                e.printStackTrace();
+                return null;
+            }
+            String queryId = (String) query(format("SELECT query_id from system.runtime.queries" +
+                    " WHERE query = '%s' ORDER BY created DESC LIMIT 1", sqlStatement)).row(0).get(0);
+            return QueryClient.cancelQuery(queryId);
+        }
+    }
+
+    private void doCancelTest(String sql)
+    {
+        Response response = null;
+        Boolean isQueryCancelled;
+        int statusCode;
+
+        ExecutorService executor = Executors.newFixedThreadPool(2);
+        Future<Boolean> executeQueryThread = executor.submit(new executeQuery(sql));
+        Future<Response> cancelQueryThread = executor.submit(new cancelQuery(sql));
+
+        try {
+            isQueryCancelled = executeQueryThread.get(1, TimeUnit.MINUTES);
+        }
```

Comment:
I'd like to see something here that verifies the query really was canceled.  Checking the return code is good, but you should also ask the server if the query is still running.